### PR TITLE
Created and installed Virtual environment

### PR DESCRIPTION
**Pull request by Dozie2001**
Added virtual environment to the dependencies so that packages installed are set in one place. A virtual environment in Python is a self-contained directory that contains its own Python interpreter and libraries, separate from the system's Python interpreter. It is a way to create isolated environments for different Python projects, each with its own dependencies, packages, and Python version. Here are some key reasons for using virtual environments:

* Isolation: Virtual environments allow you to create isolated environments for different projects. This means that the dependencies installed in one virtual environment do not affect other environments or the system's Python installation. This helps prevent conflicts between different project requirements.

* Dependency Management: When working on multiple projects, each project may have different dependencies or require specific versions of libraries. Virtual environments enable you to manage and install project-specific dependencies without interfering with other projects or the system-wide Python environment.

* Version Compatibility: Some projects may require a specific version of Python. Virtual environments make it easy to use different Python versions for different projects. This ensures that each project runs on the desired Python interpreter without affecting the system's default Python version.

* Sandboxing: Virtual environments provide a sandboxed environment where you can experiment with different packages, libraries, and configurations without affecting your system-wide Python installation. This is particularly useful for testing or trying out new packages.

* Cleaner Development and Deployment: When you share your code with others or deploy your application, using a virtual environment ensures that the required dependencies are clearly defined. This makes it easier for collaborators to set up the same environment and avoids issues related to different dependency versions. 

Also made adjustments to README to show the steps needed when setting up the virtual environment

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -86,6 +86,7 @@ ipython_config.py
 #   For a library or package, you might want to ignore these files since the code is
 #   intended to run in multiple environments; otherwise, check them in:
 # .python-version
+env
```
Comment (vantage-ola) [inline .gitignore:89]: if you check, line 122...all the possible environment files are in the .gitignore file.

File: README.md [hunk 2]
```diff
@@ -54,7 +54,18 @@ git clone git@github.com:vantage-ola/track-now.git
 git checkout -b <your branch name>
 ```
 
-3. To access the mongo db database to run the flask api , create a local environment `/backend/.env` and copy the variables.(READ_ACCESS_ONLY)
+3. Install dependencies:
+```shell
+pip3 install DEPENDENCIES.txt
```
Comment (vantage-ola) [inline README.md:59]: 100% i approve of this

File: README.md [hunk 3]
```diff
@@ -54,7 +54,18 @@ git clone git@github.com:vantage-ola/track-now.git
...
+```
+
+4. Create and activate virtualenvironment.
```
Comment (vantage-ola) [inline README.md:62]: creating a virtual env should come first before installing dependencies... this should be no 3, then installing dependencies should be no 4...you can also add installation for windows users also ...

like
  - On Windows:

     ```bash
     .\venv\Scripts\activate
     ```

   - On macOS and Linux:

     ```bash
     source venv/bin/activate
     ```

File: DEPENDENCIES.txt [hunk 4]
```diff
@@ -1,10 +1,14 @@
 blinker==1.7.0
 click==8.1.7
+distlib==0.3.8
```
Comment (vantage-ola) [inline DEPENDENCIES.txt:3]: what do these packages do?

**Review by vantage-ola — COMMENTED** [review]
good job dozie, but consider the review and make changes to them so i can approve asap!

File: .gitignore (hunk 1 above)
Comment (Dozie2001) [reply .gitignore:89]: Ok thank my bad on that

File: DEPENDENCIES.txt
Comment (Dozie2001) [inline DEPENDENCIES.txt:1]: removed package, It must have sneaked when froze virtualenv into dependencies

**Pull request merged**